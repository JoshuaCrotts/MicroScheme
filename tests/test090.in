(define (cadr l) (car (cdr l)))
(define (caddr l) (car (cdr (cdr l))))

(define ev 
    (Î» (y) (cond [(eq? (car y) '+) (+ (cadr y) (caddr y))]
                 [(eq? (car y) '-) (- (cadr y) (caddr y))]
                 [(eq? (car y) '*) (* (cadr y) (caddr y))]
                 [(eq? (car y) '/) (/ (cadr y) (caddr y))]
                 [else ("ERR invalid op")]
            )
    )
)

(ev '(+ 2 3))
(ev '(- 2 3))
(ev '(* 2 3))
(ev '(/ 2 3))
