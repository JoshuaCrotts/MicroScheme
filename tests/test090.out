5
-1
6
0.6666666666666666666666666666666666
